test/utils/interface-common-factory.js

```diff
@@ -8,10 +8,15 @@ const callbackify = require('callbackify')
 const mergeOptions = require('merge-options')
 const IPFS = require('../../src')
 
+const DEFAULT_FACTORY_OPTIONS = {
+  type: 'proc',
+  exec: IPFS
+}
+
 function createFactory (options) {
   options = options || {}
 
-  options.factoryOptions = options.factoryOptions || { type: 'proc', exec: IPFS }
+  options.factoryOptions = options.factoryOptions || DEFAULT_FACTORY_OPTIONS
```

Comment:
I don't know if ipfsd mutates these, it it does you'll get some weird things happening...

```suggestion
  options.factoryOptions = options.factoryOptions || { ...DEFAULT_FACTORY_OPTIONS }
```

Reply:
I think we don't want to spread `DEFAULT_FACTORY_OPTIONS`. If you check [`createFactory()`](https://github.com/ipfs/js-ipfs/blob/e9ae0a4d2c790a14c11142692cbce75f06f1df39/test/utils/interface-common-factory.js#L19) you will see that we either have `factoryOptions` if passed or `DEFAULT_FACTORY_OPTIONS`. I did try to keep the daemon config exactly as it was previously.

Reply:
You've pulled them out into a `const DEFAULT_FACTORY_OPTIONS` object which is possibly used multiple times. It's similar but not exactly the same as it was before.

If we call `createFactory()` with no options then `options.factoryOptions` will be `DEFAULT_FACTORY_OPTIONS` which is `{ type: 'proc', exec: IPFS }`.

If `options.factoryOptions` gets mutated later in this function or by `ipfsd-ctl` then the next call to `createFactory()` will get `{ type: 'proc', exec: IPFS, some: 'mutation' }` which are possibly no longer the intended defaults.

I'm suggesting spreading the defaults so each call to `createFactory()` gets a new set of defaults.

Reply:
Yes, you're right. I wasn't answering you considering the real problem here. I got your point, nice catch! Thanks.